src/main/java/org/opensearch/security/identity/SecuritySubject.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.identity;
+
+import java.security.Principal;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.identity.NamedPrincipal;
+import org.opensearch.identity.Subject;
+import org.opensearch.identity.tokens.AuthToken;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.security.user.User;
+
+public class SecuritySubject implements Subject {
+
+	private ThreadContext threadContext;
+
+	public SecuritySubject() {}
+
+	public void setThreadContext(ThreadContext threadContext) {
+		this.threadContext = threadContext;
+	}
+
+	@Override
+	public Principal getPrincipal() {
+		if (threadContext == null) {
+			return NamedPrincipal.UNAUTHENTICATED;
+		}
+		final User user = (User) threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+		if (user == null) {
+			return NamedPrincipal.UNAUTHENTICATED;
+		}
+		return new NamedPrincipal(user.getName());
+	}
+
+	@Override
+	public void authenticate(AuthToken authToken) {
```

Comment:
Not sure how to handle this

Reply:
@peternied With regard to our other chat about the level where this authenticate method should be, what did you have in mind for this? 

I am not sure how we are planning on handling this call but it is basically dead for the time being. I would like this to inject the new user into the thread context based on the resolution of the auth credentials. 

The issue is that the tokens should not be hitting this. I would expect authc/z to take place with the normal mechanisms for handling basic auth & jwts. In turn, there should not be any difference to the security plugin whether a on behalf of token or service account token are being used vs. a normal request with a Basic or Bearer auth header. 

So I will need to make the injection via this method standard for all requests not just those involving OBO or Service Accounts.

Reply:
Lets no-op this for the time being, we still haven't fully aligned what should happen where inside of core vs plugins as far as authentication goes.